kernel/kernel-api/src/main/java/io/delta/kernel/internal/ActionCommitLog.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal;
+
+import static io.delta.kernel.internal.DeltaErrors.*;
+import static io.delta.kernel.internal.fs.Path.getName;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.ColumnarBatch;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.exceptions.InvalidTableException;
+import io.delta.kernel.exceptions.KernelException;
+import io.delta.kernel.exceptions.TableNotFoundException;
+import io.delta.kernel.expressions.ExpressionEvaluator;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.internal.actions.AddFile;
+import io.delta.kernel.internal.actions.RemoveFile;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.replay.ActionsIterator;
+import io.delta.kernel.internal.util.FileNames;
+import io.delta.kernel.types.*;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.FileStatus;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+/**
+ * Stores code to read the raw actions within the *commit files* of the _delta_log. This is used for
+ * CDF, streaming, and more.
+ */
+public class ActionCommitLog {
+
+  /**
+   * Represents a Delta action. This is used to request which actions to read from the commit files
+   * in {@link TableImpl#getChangesByVersion(Engine, long, long, Set)}.
+   *
+   * <p>See the Delta protocol for more details
+   * https://github.com/delta-io/delta/blob/master/PROTOCOL.md#actions
+   */
+  public enum DeltaAction {
+    // TODO do we want to provide an API that can evolve the action schema w/out release?
```

Comment:
what's your plan to resolve this open TODO? do we need to meet and discuss?

Reply:
Removing this for now, I will follow up on this separately (I've added to my working doc)